server/mqtt_test.go

```diff
@@ -7633,3 +7737,34 @@ func BenchmarkMQTT_QoS1_PubSub2_256b_Payload(b *testing.B) {
 func BenchmarkMQTT_QoS1_PubSub2___1K_Payload(b *testing.B) {
 	mqttBenchPubQoS1(b, mqttPubSubj, sizedString(1024), 2)
 }
+
+func BenchmarkMQTTDoNotDeliverSubject(b *testing.B) {
+	buf := make([]byte, 32)
+	jsSubjects := make([]string, 100)
+	mqttSubjects := make([]string, 100)
+	noSubjects := make([]string, 1000)
+
+	for i := 0; i < 100; i++ {
+		_, _ = rand.Read(buf)
```

Comment:
I don't think using rand here is a good thing because it could generate invalid subject, no? what is there something like "ancd..dwre"? So I would revert the mathrand change and go with something actually less random here (which also will add reproducibility in the bench)

Reply:
There's a 1000+ of them there, so it shouldn't be much variance? And we don't look beyond 8 bytes anyway? But yes, I see your point. I can just do a handful of static strings, sure.

Reply:
Alternatively just hex-encode or something to avoid invalid characters.